### PR TITLE
Creating_a_thumbnail

### PR DESCRIPTION
**Pull request by JvSdv**
# Description
Create a thumbnail for the video using the PRAW library.

- thumbnail : Need to put the thumbnail.png image in the assets/backgrounds folder.
- font_family : The font family to use for the title.
- font_size : The font size to use for the title.
- font_color : The font color to use for the title.
- width,height : The width of the thumbnail is based on the thumbnail image.

1. Thumbnail created takes the image of the assets and writes the title post on top of this image.
2. The thumbnail is then saved in the assets/{id}/temp folder.
3. And finally the thumbnail is saved in the results/{subreddit}/thumbnails folder.

# Issue Fixes
None

# Checklist:

- [x] I am pushing changes to the **develop** branch
- [x] I am using the recommended development environment
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have formatted and linted my code using python-black and pylint
- [x] I have cleaned up unnecessary files
- [x] My changes generate no new warnings
- [x] My changes follow the existing code-style
- [x] My changes are relevant to the project

# Any other information (e.g how to test the changes)
Thanks for the bot helps me bring good content to Brazil

### CODE REVIEW

**Comment by JvSdv** [comment]
Put any image in assets/backgrounds folder
thumbnail_large
thumbnail_shorts

**Review by OpenSourceSimon — COMMENTED** [review]
Looks good to me, haven't tested it though.

**Review by OpenSourceSimon — APPROVED** [review]

**Comment by JasonLovesDoggo** [comment]
I would just change this to drop an image in the folder and it will use that as the background

**Review by JasonLovesDoggo — CHANGES_REQUESTED** [review]
See above comment

**Comment by OpenSourceSimon** [comment]
> I would just change this to drop an image in the folder and it will use that as the background 

That's possible if you put a thumbnail.png in the assets/backgrounds/ folder. Or do you mean something else?

**Comment by JvSdv** [comment]
That's right. maybe change the name to background.png or something else?

**Comment by OpenSourceSimon** [comment]
> That's right. maybe change the name to background.png or something else?
> 
> 

I think it shouldn't be in the assets folder since it's only for media generated by the bot. Maybe utils?

File: utils/.config.template.toml [hunk 1]
```diff
@@ -28,7 +28,10 @@ storymode = { optional = true, type = "bool", default = false, example = false,
...
 #background_audio = { optional = true, type = "bool", default = false, example = false, options = [true, false,], explanation = "Sets a audio to play in the background (put a background.mp3 file in the assets/backgrounds directory for it to be used.)" }
 #background_audio_volume = { optional = true, type = "float", default = 0.3, example = 0.1, explanation="Sets the volume of the background audio. only used if the background_audio is also set to true" }
-
+background_thumbnail = { optional = true, type = "bool", default = false, example = false, options = [true, false,], explanation = "Sets a thumbnail to the video (put a thumbnail.png file in the assets/backgrounds directory for it to be used.)" }
+background_thumbnail_font_family = { optional = true, default = "arial", example = "arial", explanation = "Sets the font family for the thumbnail text" }
+background_thumbnail_font_size = { optional = true, type = "int", default = 96, example = 96, explanation = "Sets the font size in pixels for the thumbnail text" }
+background_thumbnail_font_color = { optional = true, default = "255,255,255", example = "255,255,255", explanation = "Sets the font color for the thumbnail text" }
```
Comment (JasonLovesDoggo) [inline utils/.config.template.toml:35]: Explain that it's a rgb color

File: video_creation/final_video.py [hunk 2]
```diff
@@ -124,14 +127,45 @@ def make_final_video(
     final = CompositeVideoClip([background_clip, image_concat])
     title = re.sub(r"[^\w\s-]", "", reddit_obj["thread_title"])
     idx = re.sub(r"[^\w\s-]", "", reddit_obj["thread_id"])
+    title_thumb = reddit_obj["thread_title"]
 
-    filename = f"{name_normalize(title)[:251]}.mp4"
```
Comment (JasonLovesDoggo) [inline video_creation/final_video.py:128]: Why.... 251 was specific..

**Review by JasonLovesDoggo — CHANGES_REQUESTED** [review]

**Comment by JasonLovesDoggo** [comment]
> > That's right. maybe change the name to background.png or something else?
> > 
> > 
> 
> I think it shouldn't be in the assets folder since it's only for media generated by the bot. Maybe utils?

This is generated by the bot though??

**Comment by OpenSourceSimon** [comment]
> > > That's right. maybe change the name to background.png or something else?
> > > 
> > > 
> > 
> > I think it shouldn't be in the assets folder since it's only for media generated by the bot. Maybe utils?
> 
> This is generated by the bot though?? 

Sorry for not being clear. I was talking about the input background image. The assets folder is mostly for temp files and can be easily removed for bug fixing. The end thumbnail will go at the results folder if I remember correctly. 

But now there are more PRs with customization and they're using the assets folder too so we'll keep it also in this PR.

**Comment by JasonLovesDoggo** [comment]
> > > > That's right. maybe change the name to background.png or something else?
> > > > 
> > > > 
> > > 
> > > I think it shouldn't be in the assets folder since it's only for media generated by the bot. Maybe utils?
> > 
> > This is generated by the bot though?? 
> 
> Sorry for not being clear. I was talking about the input background image. The assets folder is mostly for temp files and can be easily removed for bug fixing. The end thumbnail will go at the results folder if I remember correctly. 
> 
> But now there are more PRs with customization and they're using the assets folder too so we'll keep it also in this PR.

From what I understand about this MR, the background image is generated by the bot and yeah it should definitely end up in results if it's going to be an output

File: video_creation/final_video.py [hunk 3]
```diff
@@ -124,14 +127,45 @@ def make_final_video(
...
+    filename = f"{name_normalize(title)[:150]}"
```
Comment (OpenSourceSimon) [inline video_creation/final_video.py:132]:
```suggestion
    filename = f"{name_normalize(title)[:251]}"
```
On suggestion of Jason, 251 as a limit was specifically chosen.

**Review by OpenSourceSimon — APPROVED** [review]
Tested and looks good! Thank you for your PR!

**Pull request merged**